src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -550,6 +576,35 @@ def batchComplete(self, result):
         msg = "Fitting completed successfully in: %s s.\n" % GuiUtils.formatNumber(elapsed)
         self.parent.communicate.statusBarUpdateSignal.emit(msg)
 
+    def outputConstraintError(self, exception):
+        """
+        Returns an error string highlighting the error in the constraint.
+        The argument is an exception that was raised by expression.py, which are interpreted as following:
+        CyclicDefinitionError: parameter has cyclic dependencies
+        UnknownParameterNameError: unknown symbol found in constraint
+        ConstraintSyntaxError: syntax problem such as missing parentheses
+        """
+        assert(isinstance(exception, Exception))
+        if isinstance(exception, CyclicDefinitionError):
+            error_message = "Cyclic dependency with parameter: <b>%s</b>"%(exception.parameter)
+
+        elif isinstance(exception, UnknownParameterNameError):
+            constraint = exception.constraint
+            for symbol in exception.unknown_symbol:
+                constraint = constraint.replace(symbol, "<b>%s</b>" % (symbol))
```

Comment:
`unknown_symbol` is a string containing a comma separated list of symbols, so this loops over the individual characters.

`replace` does not respect word boundaries, so if the unknown symbol is 'radius', `replace` will replace it inside 'M1.radius_polar'.

You can use `re.sub` instead.  Unfortunately, you cannot use the `\b` word boundary marker since symbols can contain periods in addition to alphanumeric characters, so you need to use the more awkward:
```python
# Identify the symbol for replacement as everything between a word boundary. Since symbols
# can contain '.', we need to use negative lookbehind and negative lookahead on a character set
# containing '.' to check for the boundaries.  Also, if there is a '.' in the symbol it could match any
# character, so it needs to be turned into a regular expression that matches '.'.
pattern = f"(?<![a-zA-Z0-9_.]){symbol.replace('.', '[.]')}(?![a-zA-Z0-9_.])"
re.sub(pattern, f"<b>{symbol}</b>")
```